### PR TITLE
Added extended euclidean algorithm in java

### LINKED ISSUE #2906: Extended Euclidean Algorithm
- [x] C
- [x] C++
- [x] C#
- [x] CoffeeScript
- [ ] Dart - @Jitmanew-Tyagi
- [ ] Go
- [x] Java - @lakshyajit165 
- [x] JS - @Satyam3976
- [ ] Kotlin - @Manish-cloud
- [x] Php - @utkarshanehe
- [x] Python
- [ ] Readme - @Amitsharma45
- [x] Ruby - @atarax665
- [ ] Rust
- [x] TypeScript - @shivanidalmia1203

### PR DESCRIPTION
**Pull request by lakshyajit165**
Fixes #2906 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Implemented the extended euclidean algorithm in Java

#### Languages Used:
- Java

#### Files Added:
- Extended_Euclidean_Algorithm.java

### CODE REVIEW

**Comment by lakshyajit165** [comment]
@asha15 please review my work.

File: Extended_Euclidean_Algorithm/Extended_Euclidean_Algorithm.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+        // input a = 35, b = 15; 
+        int a = s.nextInt(); 
+        int b = s.nextInt();
+
+        int g = getGCD(a, b, x, y); // Output: GCD of 35 , 15 = 5
+        System.out.print("GCD of" + a +  " , " + b+ " = " + g); 
+  
+    } 
+}
```
Comment (atarax665) [inline Extended_Euclidean_Algorithm/Extended_Euclidean_Algorithm.java:40]: Do add sample Input/Output at the end of the program.

File: Extended_Euclidean_Algorithm/Extended_Euclidean_Algorithm.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        // Base Case 
+        if (a == 0) 
+        { 
+            x = 0; 
+            y = 1; 
+            return b; 
+        } 
+  
+        int x1=1, y1=1; // To store results of recursive call 
+        int gcd = getGCD(b%a, a, x1, y1);
```
Comment (atarax665) [inline Extended_Euclidean_Algorithm/Extended_Euclidean_Algorithm.java:17]: Use proper spacing between operators (i.e: b % a)

File: Extended_Euclidean_Algorithm/Extended_Euclidean_Algorithm.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        x = y1 - (b/a) * x1; 
+        y = x1; 
+  
+        return gcd; 
+    } 
+  
+// Main function
+    public static void main(String[] args) 
+    { 
+        int x=1, y=1;
```
Comment (atarax665) [inline Extended_Euclidean_Algorithm/Extended_Euclidean_Algorithm.java:29]: x = 1, y = 1;

File: Extended_Euclidean_Algorithm/Extended_Euclidean_Algorithm.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    { 
+        int x=1, y=1; 
+        Scanner s = new Scanner(System.in);
+
+        // input a = 35, b = 15; 
+        int a = s.nextInt(); 
+        int b = s.nextInt();
+
+        int g = getGCD(a, b, x, y); // Output: GCD of 35 , 15 = 5
+        System.out.print("GCD of" + a +  " , " + b+ " = " + g);
```
Comment (atarax665) [inline Extended_Euclidean_Algorithm/Extended_Euclidean_Algorithm.java:37]: System.out.print("GCD of " + a +  " , " + b + " = " + g);

**Comment by lakshyajit165** [comment]
@atarax665 and @asha15. I have pushed the modifications to my code.
Please review.

File: Extended_Euclidean_Algorithm/Extended_Euclidean_Algorithm.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        int x = 1, y = 1; 
+        Scanner s = new Scanner(System.in);
+
+        
+        int a = s.nextInt(); 
+        int b = s.nextInt();
+
+        int g = getGCD(a, b, x, y);
+        System.out.print("GCD of " + a + " , " + b + " = " + g);
+
```
Comment (atarax665) [inline Extended_Euclidean_Algorithm/Extended_Euclidean_Algorithm.java:38]: Remove all the unnecessary blank lines in main function.

Comment (lakshyajit165) [reply Extended_Euclidean_Algorithm/Extended_Euclidean_Algorithm.java]: @atarax665 I have made the changes. Please review.

File: Extended_Euclidean_Algorithm/Extended_Euclidean_Algorithm.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        int g = getGCD(a, b, x, y);
+        System.out.print("GCD of " + a + " , " + b + " = " + g);
+   } 
+} 
+
+// INPUT
+// a = 35, b = 15; 
+
+// Output: 
+// GCD of 35 , 15 = 5
```
Comment (singh-shreya6) [inline Extended_Euclidean_Algorithm/Extended_Euclidean_Algorithm.java:42]: Please add a blank line at end of file.

File: Extended_Euclidean_Algorithm/Extended_Euclidean_Algorithm.java [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+        int x = 1, y = 1; 
+        Scanner s = new Scanner(System.in);
+        int a = s.nextInt(); 
+        int b = s.nextInt();
+        int g = getGCD(a, b, x, y);
+        System.out.print("GCD of " + a + " , " + b + " = " + g);
+   } 
+} 
+
+// INPUT
```
Comment (singh-shreya6) [inline Extended_Euclidean_Algorithm/Extended_Euclidean_Algorithm.java:38]: Please convert input/output to multi line comment.

File: Extended_Euclidean_Algorithm/Extended_Euclidean_Algorithm.java [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+
+/*
+   INPUT
+   a = 35, b = 15; 
+
+   Output: 
+   GCD of 35 , 15 = 5
+*/
```
Comment (singh-shreya6) [inline Extended_Euclidean_Algorithm/Extended_Euclidean_Algorithm.java:45]: You didn't add a blank line after */?

**Pull request merged**